### DISCUSSION TITLE
Issue updating Realm pods dependency

### DISCUSSION THREAD

**Discussion by asoneji** [renovatebot/renovate#21689] [Suggest an Idea]
**What Renovate type are you using?**
GitHub

**Describe the bug**
Issue updating Cocoapods dependency

**Did you see anything helpful in debug logs?**
   * [Artifact file update failure](https://github.com/asoneji/sample-realm-ios-project-cocoapods-renovate-demo/pull/2#issuecomment-649812927)
   * Job log [here](https://app.renovatebot.com/dashboard#github/asoneji/sample-realm-ios-project-cocoapods-renovate-demo/194341121)

**To Reproduce**
1. Created sample repo [sample-realm-ios-project-cocoapods-renovate-demo](https://github.com/asoneji/sample-realm-ios-project-cocoapods-renovate-demo.git)
2. Create a sample XCode Project [sample-ios-project](https://github.com/asoneji/sample-realm-ios-project-cocoapods-renovate-demo/tree/master/sample-ios-project)
3. Did `pod init` and added a pod dependency
   * Added at top:
     * `platform :ios, '11.2'`
   * Added Dependency:
     * `pod 'Analytics', '3.8.1'`
4. Connect Renovate using [renovate](https://github.com/marketplace/renovate) to the repo [sample-realm-ios-project-cocoapods-renovate-demo](https://github.com/asoneji/sample-realm-ios-project-cocoapods-renovate-demo).
5. Renovate created a [pr 1](https://github.com/asoneji/sample-realm-ios-project-cocoapods-renovate-demo/pull/1) with renovate config which I merged.
6. Renovate created a [pr 2](https://github.com/asoneji/sample-realm-ios-project-cocoapods-renovate-demo/pull/2) to Update dependency Analytics 3.8.1 to 3.9.0.
   * Successfully dependency update for renovate bot.
   * Job log [here](https://app.renovatebot.com/dashboard#github/asoneji/sample-realm-ios-project-cocoapods-renovate-demo/195262683)
7. Add Realm
   * Added Dependency:
     * `pod 'RealmSwift', '5.0.3'`
8. Renovate created a [pr 4](https://github.com/asoneji/sample-realm-ios-project-cocoapods-renovate-demo/pull/4) to Update dependency RealmSwift 5.0.3 to 5.1.0.
   * [Artifact file update failure in PR 4](https://github.com/asoneji/sample-ios-project-cocoapods-renovate-debug-source/pull/4#issuecomment-650465262)
   * [Artifact file update failure in PR 2](https://github.com/asoneji/sample-ios-project-cocoapods-renovate-debug-source/pull/2#issuecomment-650465215)
   * Job log [here](https://app.renovatebot.com/dashboard#github/asoneji/sample-realm-ios-project-cocoapods-renovate-demo/195262683)

**Note**: If I remove `RealmSwift` from podfile the [pr 2](https://github.com/asoneji/sample-realm-ios-project-cocoapods-renovate-demo/pull/2) again will behave successfully dependency update for renovate bot. Adding realm impacts all PR's.

**Additional context**

**Comment by rarkins** [comment]
An Artifact failure *alone* is not proof on its own that there's a bug. Are the cocoapods commands that are run by Renovate correct? (you should be able to retrieve them from the log). Do you get a different result if you run the same commands locally? etc.

And is this just realmswift or do you get the same error for every dependency?

**Comment by asoneji** [comment]
Cocoapods commands that are run by Renovate seem correct to me.
```
{
  "command": "docker run --rm --name=renovate_cocoapods --label=renovate_child -v \"/mnt/renovate/gh/asoneji/sample-realm-ios-project-cocoapods-renovate-demo\":\"/mnt/renovate/gh/asoneji/sample-realm-ios-project-cocoapods-renovate-demo\" -v \"/tmp/renovate-cache\":\"/tmp/renovate-cache\" -e CP_HOME_DIR -w \"/mnt/renovate/gh/asoneji/sample-realm-ios-project-cocoapods-renovate-demo/sample-ios-project\" renovate/cocoapods:1.9.1 bash -l -c \"pod install\""
}
```

The same commands `pod install` works correctly locally.

So far I have noticed the issue only with `realmswift`

**Comment by asoneji** [comment]
I added analytics pod dependency to this project and notice that also failed here https://github.com/asoneji/sample-realm-ios-project-cocoapods-renovate-demo/pull/3 but the same pod worked in another project https://github.com/asoneji/sample-ios-project-cocoapods-test-renovatebot/pull/2

**Comment by asoneji** [comment]
Sorry my fault. In the repo I created to reproduce the realm issue I missed the source url in the Podfile. I have added that will checked the rebase options in the pr's.

**Comment by asoneji** [comment]
Let me try few things and update the issue with correct information. I will hopefully updated by tomorrow.

**Comment by asoneji** [comment]
I found a few issues and have reproducible steps and will update this issue and add create other issues with details sometime next week.

- Will update this issue with Realm pod issue
- Adding source in Podfile creates an issue - will create new issue for this

**Comment by asoneji** [comment]
1. Created sample repo [sample-realm-ios-project-cocoapods-renovate-demo](https://github.com/asoneji/sample-realm-ios-project-cocoapods-renovate-demo.git)
2. Create a sample XCode Project [sample-ios-project](https://github.com/asoneji/sample-realm-ios-project-cocoapods-renovate-demo/tree/master/sample-ios-project)
3. Did `pod init` and added a pod dependency
   * Added at top:
     * `platform :ios, '11.2'`
   * Added Dependency:
     * `pod 'Analytics', '3.8.1'`
4. Connect Renovate using [renovate](https://github.com/marketplace/renovate) to the repo [sample-realm-ios-project-cocoapods-renovate-demo](https://github.com/asoneji/sample-realm-ios-project-cocoapods-renovate-demo).
5. Renovate created a [pr 1](https://github.com/asoneji/sample-realm-ios-project-cocoapods-renovate-demo/pull/1) with renovate config which I merged.
6. Renovate created a [pr 2](https://github.com/asoneji/sample-realm-ios-project-cocoapods-renovate-demo/pull/2) to Update dependency Analytics 3.8.1 to 3.9.0.
   * Successfully dependency update for renovate bot.
   * Job log [here](https://app.renovatebot.com/dashboard#github/asoneji/sample-realm-ios-project-cocoapods-renovate-demo/195262683)
7. Add Realm
   * Added Dependency:
     * `pod 'RealmSwift', '5.0.3'`
8. Renovate created a [pr 4](https://github.com/asoneji/sample-realm-ios-project-cocoapods-renovate-demo/pull/4) to Update dependency RealmSwift 5.0.3 to 5.1.0.
   * [Artifact file update failure in PR 4](https://github.com/asoneji/sample-realm-ios-project-cocoapods-renovate-demo/pull/4#issuecomment-650665632)
   * [Artifact file update failure in PR 2](https://github.com/asoneji/sample-realm-ios-project-cocoapods-renovate-demo/pull/2#issuecomment-650653777)
   * Job log [here](https://app.renovatebot.com/dashboard#github/asoneji/sample-realm-ios-project-cocoapods-renovate-demo/195262683)


**Note**: If I remove `RealmSwift` from podfile the [pr 2](https://github.com/asoneji/sample-realm-ios-project-cocoapods-renovate-demo/pull/2) again will behave successfully dependency update for renovate bot. Adding realm impacts all PR's.

**Comment by asoneji** [comment]
https://github.com/renovatebot/renovate/issues/6613 -  Adding source to Podfile breaks cocoapods dependency

**Comment by rarkins** [comment]
Are these two separate problems? Can you rename this issue if so and identify if one issue depends on the other?

And is this definitely a Renovate problem and not a `pod` problem?

**Comment by rarkins** [comment]
@asoneji can you double check those links? `https://github.com/asoneji/sample-ios-project-cocoapods-renovate-debug-source/issues/4` is a 404 for example

**Comment by asoneji** [comment]
> Are these two separate problems? Can you rename this issue if so and identify if one issue depends on the other?

Yes, they are separate problems. I have renamed the issue. They are separate issues and don't have a dependency on one or other.


> And is this definitely a Renovate problem and not a `pod` problem?

It works fine locally and in our build machine so to me it seems to be a Renovate issue. I would guess it has something to do whit renovate docker image as [Realm](https://realm.io/docs/swift/latest/) has prerequisites of `XCode 10.0 or higher`.


> @asoneji can you double check those links? `https://github.com/asoneji/sample-ios-project-cocoapods-renovate-debug-source/issues/4` is a 404 for example

Sorry about that update the link to https://github.com/asoneji/sample-realm-ios-project-cocoapods-renovate-demo/pull/4

**Comment by viceice** [comment]
Ok, so we might be missing a prerequisite in our cocoapods docker images. We don't have xcode there, do we need it for this simple step? ☹️

**Comment by asoneji** [comment]
I think not all pod require XCode but I guess some due.

**Comment by viceice** [comment]
@asoneji can you give me maintainer rights to your test repo, so I can see the renovate logs.

It's not possible to install xcode on Linux, but maybe we just need a swift compiler

**Comment by asoneji** [comment]
> @asoneji can you give me maintainer rights to your test repo, so I can see the renovate logs.
>
 done gave @viceice and @rarkins access. Feel free to change things as the step to reproduce are documented and easy to recreate.
> It's not possible to install xcode on Linux, but maybe we just need a swift compiler
>
Is it possible to have a mac os docker image or something that is xcode compatible?



If the Realm pr fail why does it impact all the other pr’s? Can this be fixed so only the impacted (realm) pr is broken?

**Comment by rarkins** [comment]
If the presence of realm causes "pod install" to fail then it will cause all PRs to fail because all need the same pod command.

**Comment by viceice** [comment]
ok, i can reproduce it locally:

```
#24 11.77 -> Installing Realm (5.0.3)
#24 11.77  > Git download
#24 11.77  > Git download
#24 11.77      $ /usr/bin/git clone https://github.com/realm/realm-cocoa.git /tmp/d20200630-14-jb9y84 --template= --single-branch --depth 1 --branch v5.0.3
#24 11.77      Cloning into '/tmp/d20200630-14-jb9y84'...
#24 14.07      Note: switching to 'd35c699577de02ffb2e025e9e6b308a38408bfe6'.
#24 14.07
#24 14.07      You are in 'detached HEAD' state. You can look around, make experimental
#24 14.07      changes and commit them, and you can discard any commits you make in this
#24 14.07      state without impacting any branches by switching back to a branch.
#24 14.07
#24 14.07      If you want to create a new branch to retain commits you create, you may
#24 14.07      do so (now or later) by using -c with the switch command. Example:
#24 14.07
#24 14.07        git switch -c <new-branch-name>
#24 14.07
#24 14.07      Or undo this operation with:
#24 14.07
#24 14.07        git switch -
#24 14.07
#24 14.07      Turn off this advice by setting config variable advice.detachedHead to false
#24 14.07
#24 14.13      $ /usr/bin/git -C /tmp/d20200630-14-jb9y84 submodule update --init --recursive
#24 14.14      Submodule 'Realm/ObjectStore' (https://github.com/realm/realm-object-store.git) registered for path 'Realm/ObjectStore'
#24 14.15      Cloning into '/tmp/d20200630-14-jb9y84/Realm/ObjectStore'...
#24 18.05      Submodule 'external/catch' (https://github.com/catchorg/Catch2) registered for path 'Realm/ObjectStore/external/catch'
#24 18.05      Cloning into '/tmp/d20200630-14-jb9y84/Realm/ObjectStore/external/catch'...
#24 22.04      Submodule path 'Realm/ObjectStore': checked out '995bd9ba83a7db177e3332e354c601fa5fc28b7f'
#24 22.04      Submodule path 'Realm/ObjectStore/external/catch': checked out 'd10b9bd02e098476670f5eb0527d2c7281476e8a'
#24 22.04  > Running prepare command
#24 22.04    $ /bin/bash -c  set -e sh build.sh cocoapods-setup
#24 22.04    build.sh: 13: set: Illegal option -o pipefail
#24 22.11 [!] /bin/bash -c
#24 22.11 set -e
#24 22.11 sh build.sh cocoapods-setup
#24 22.11
#24 22.11 build.sh: 13: set: Illegal option -o pipefail
```

**Comment by rarkins** [comment]
Similar problems: https://github.com/realm/realm-cocoa/issues/2662

**Comment by viceice** [comment]
after running  `ln -sf /bin/bash /bin/sh` we get the following error:
```
#25 19.86  > Running prepare command
#25 19.86    $ /bin/bash -c  set -e sh build.sh cocoapods-setup
#25 19.87    getconf: Unrecognized variable `DARWIN_USER_TEMP_DIR'
#25 19.87    ./scripts/swift-version.sh: line 123: xcode-select: command not found
#25 19.94 [!] /bin/bash -c
#25 19.94 set -e
#25 19.94 sh build.sh cocoapods-setup
#25 19.94
#25 19.94 getconf: Unrecognized variable `DARWIN_USER_TEMP_DIR'
#25 19.94 ./scripts/swift-version.sh: line 123: xcode-select: command not found
```

https://github.com/realm/realm-cocoa/issues/3263 No linux support for `realm`, so we can't do anything here

**Comment by rarkins** [comment]
Ideally this type of stuff shouldn't be necessary just to update the *lockfile*. I wonder if/where we can bypass such scripts. e.g. maybe a new `--ignore-scripts` option for `pod` ?

**Comment by viceice** [comment]
Maybe we can use this https://github.com/facebook/xcbuild to fake xcode 🤔

**Comment by rarkins** [comment]
![season 3 netflix - 3o6wr9xUAzjEXcA5BS](https://user-images.githubusercontent.com/6311784/86109262-7820e700-bac4-11ea-89e0-8ca42bdd6bbb.gif)

**Comment by viceice** [comment]
https://github.com/realm/realm-cocoa/blob/master/scripts/swift-version.sh Looks like it directly calls xcode, so we would need to completely fake xcode. That would be a lot work and i'm not sure if we can get it to work. 🤔

**Comment by rarkins** [comment]
Hard alternative: would a cocoapods plugin work? e.g. that mostly replicates what "install" does but skip the bits that are failing. Cocoapods' policy is not to "feature creep" stuff that doesn't benefit 80%, so they may be unlikely to accept a new `--ignore-scripts` type of option to `pod install`.

Worst but possible option: updating the Podfile.lock directly. It's not *too* complicated:

```
PODS:
  - Analytics (3.8.1)
  - Realm (5.0.3):
    - Realm/Headers (= 5.0.3)
  - Realm/Headers (5.0.3)
  - RealmSwift (5.0.3):
    - Realm (= 5.0.3)

DEPENDENCIES:
  - Analytics (= 3.8.1)
  - RealmSwift (= 5.0.3)

SPEC REPOS:
  trunk:
    - Analytics
    - Realm
    - RealmSwift

SPEC CHECKSUMS:
  Analytics: 45b1faec509367a99c7b788cc5389180c43fe3fc
  Realm: bfca1699b61b0b17c3a69ae0e648314eae91fbdb
  RealmSwift: 493c9f089cd3893b3959007973c0e4f640906ba0

PODFILE CHECKSUM: 77f78afbd3dead8f8d38bc8f1714736413da8c46

COCOAPODS: 1.9.1
```

**Comment by viceice** [comment]
We would need to check all transitive deps:
```
PODS:
  - CPDAcknowledgements (0.5.0)
  - Expecta (1.0.5)
  - FLKAutoLayout (0.2.1)
  - IRFEmojiCheatSheet (0.3.0)
  - OCHamcrest (4.3.0)
  - OCMockito (1.4.0):
    - OCHamcrest (~> 4.0)
  - ORStackView (3.0.1):
    - FLKAutoLayout (~> 0.2)
  - Specta (1.0.5)

DEPENDENCIES:
  - CPDAcknowledgements (from `../CPDAcknowledgements.podspec`)
  - Expecta (~> 1.0)
  - IRFEmojiCheatSheet
  - OCMockito (~> 1.0)
  - ORStackView
  - Specta (~> 1.0)

EXTERNAL SOURCES:
  CPDAcknowledgements:
    :path: "../CPDAcknowledgements.podspec"

SPEC CHECKSUMS:
  CPDAcknowledgements: 32f6dfcc35eed5d9897f13947024b9d32133e2fa
  Expecta: e1c022fcd33910b6be89c291d2775b3fe27a89fe
  FLKAutoLayout: 9db6b30c2008d230da608e62c607b11c23b942e6
  IRFEmojiCheatSheet: 364b2733c4e37c65ef9f56225246cdf42068370f
  OCHamcrest: cd63d27f48a266d4412c0b295b01b8f0940efa81
  OCMockito: 4981140c9a9ec06c31af40f636e3c0f25f27e6b2
  ORStackView: a1bb52748cd0ae29891c140baf22ff8972fb363c
  Specta: ac94d110b865115fe60ff2c6d7281053c6f8e8a2

PODFILE CHECKSUM: 1c110034e6fc846de3ca560c07d3c7cb43780d13

COCOAPODS: 1.0.0.beta.6
© 2020 GitHub, Inc.
```

So maybe it would be easier to `patch` or `plugin` cocoapods to add the `--ignore-scripts` parameter. We can ask the authors if adding such an option would be acceptable.

**Comment by viceice** [comment]
if i patch cocoapods to skip scripts i get the following error:


```
Errno::ENOENT - No such file or directory @ realpath_rec - ~/src/sample-realm-ios-project-cocoapods-renovate-demo/sample-ios-project/Pods/Realm/Realm/RLMAccessor.h
~/.gem/ruby/2.5.0/gems/cocoapods-1.9.1/lib/cocoapods/project.rb:326:in `realpath'
~/.gem/ruby/2.5.0/gems/cocoapods-1.9.1/lib/cocoapods/project.rb:326:in `realpath'
~/.gem/ruby/2.5.0/gems/cocoapods-1.9.1/lib/cocoapods/project.rb:326:in `reference_for_path'
~/.gem/ruby/2.5.0/gems/cocoapods-1.9.1/lib/cocoapods/project.rb:269:in `add_file_reference'
~/.gem/ruby/2.5.0/gems/cocoapods-1.9.1/lib/cocoapods/installer/xcode/pods_project_generator/file_references_installer.rb:223:in `block (2 levels) in add_file_accessors_paths_to_pods_group'
~/.gem/ruby/2.5.0/gems/cocoapods-1.9.1/lib/cocoapods/installer/xcode/pods_project_generator/file_references_installer.rb:222:in `each'
~/.gem/ruby/2.5.0/gems/cocoapods-1.9.1/lib/cocoapods/installer/xcode/pods_project_generator/file_references_installer.rb:222:in `block in add_file_accessors_paths_to_pods_group'
~/.gem/ruby/2.5.0/gems/cocoapods-1.9.1/lib/cocoapods/installer/xcode/pods_project_generator/file_references_installer.rb:213:in `each'
~/.gem/ruby/2.5.0/gems/cocoapods-1.9.1/lib/cocoapods/installer/xcode/pods_project_generator/file_references_installer.rb:213:in `add_file_accessors_paths_to_pods_group'
~/.gem/ruby/2.5.0/gems/cocoapods-1.9.1/lib/cocoapods/installer/xcode/pods_project_generator/file_references_installer.rb:96:in `block in add_source_files_references'
~/.gem/ruby/2.5.0/gems/cocoapods-1.9.1/lib/cocoapods/user_interface.rb:145:in `message'
~/.gem/ruby/2.5.0/gems/cocoapods-1.9.1/lib/cocoapods/installer/xcode/pods_project_generator/file_references_installer.rb:95:in `add_source_files_references'
~/.gem/ruby/2.5.0/gems/cocoapods-1.9.1/lib/cocoapods/installer/xcode/pods_project_generator/file_references_installer.rb:46:in `install!'
~/.gem/ruby/2.5.0/gems/cocoapods-1.9.1/lib/cocoapods/installer/xcode/pods_project_generator.rb:101:in `block in install_file_references'
~/.gem/ruby/2.5.0/gems/cocoapods-1.9.1/lib/cocoapods/user_interface.rb:145:in `message'
~/.gem/ruby/2.5.0/gems/cocoapods-1.9.1/lib/cocoapods/installer/xcode/pods_project_generator.rb:99:in `install_file_references'
~/.gem/ruby/2.5.0/gems/cocoapods-1.9.1/lib/cocoapods/installer/xcode/single_pods_project_generator.rb:17:in `generate!'
~/.gem/ruby/2.5.0/gems/cocoapods-1.9.1/lib/cocoapods/installer.rb:308:in `block in create_and_save_projects'
~/.gem/ruby/2.5.0/gems/cocoapods-1.9.1/lib/cocoapods/user_interface.rb:64:in `section'
~/.gem/ruby/2.5.0/gems/cocoapods-1.9.1/lib/cocoapods/installer.rb:303:in `create_and_save_projects'
~/.gem/ruby/2.5.0/gems/cocoapods-1.9.1/lib/cocoapods/installer.rb:295:in `generate_pods_project'
~/.gem/ruby/2.5.0/gems/cocoapods-1.9.1/lib/cocoapods/installer.rb:174:in `integrate'
~/.gem/ruby/2.5.0/gems/cocoapods-1.9.1/lib/cocoapods/installer.rb:162:in `install!'
~/.gem/ruby/2.5.0/gems/cocoapods-1.9.1/lib/cocoapods/command/install.rb:52:in `run'
~/.gem/ruby/2.5.0/gems/claide-1.0.3/lib/claide/command.rb:334:in `run'
~/.gem/ruby/2.5.0/gems/cocoapods-1.9.1/lib/cocoapods/command.rb:52:in `run'
~/.gem/ruby/2.5.0/gems/cocoapods-1.9.1/bin/pod:55:in `<top (required)>'
~/.gem/ruby/2.5.0/bin/pod:23:in `load'
~/.gem/ruby/2.5.0/bin/pod:23:in `<main>'
```
so it looks like we can't do anything here to support `realm` pod

**Comment by viceice** [comment]
This would help but closed unresolved https://github.com/CocoaPods/CocoaPods/issues/8579 😞

**Comment by asoneji** [comment]
Is it possible to have a mac os docker image or something that is xcode compatible?

**Comment by viceice** [comment]
Nope, xcode only runs on macos, it's also not allowed to run macos in docker or vm on non macos host

**Comment by viceice** [comment]
What you can do is run the renovate CLI on a Mac your own or on GitHub action and install dependencies global.

**Comment by asoneji** [comment]
I can try that for my local debugging but will not be a hosted solution.

**Comment by asoneji** [comment]
I tried Renovate CLI from my mac machine and I was able to install analytic and realm (the one that requires xcode). Here is the sample project using that: https://github.com/asoneji/sample-ios-project-cocoapods-renovate-cli-demo

I wonder how many other pod's require XCode similar to Realm.

For us we can potentially look at hosting renovate on Mac OS machine, I doubt we will be able to do this any time soon.

I wonder how the Renovate GitHub app can solve this problem? (Would be great to have a Mac OS docker image this problem would be solved in that case.)

**Comment by wilau2** [comment]
I do have the same problem with glog pod. 
```
error: C compiler cannot create executables\nSee `config.log' for more details\n
```

https://stackoverflow.com/questions/53028367/glog-configure-warning-missing-script-is-too-old-or-missing

Potential solution: 

~Instead of `pod install` I will list the dependencies I care about and run individually
`pod update <podname>`
As suggested here: https://github.com/CocoaPods/CocoaPods/issues/3294~

**Comment by viceice** [comment]
@wilau2 We can try this, but that does not work for `realm` because even `pod update <xxx>` will download all pods and execute the install scripts, which fails on non macos.

**Comment by rarkins** [comment]
Wondering if there's an external service (including GitHub Actions) that we could utilise for this particular update. Ie push the Podfile and lock, run the command, then pull the results.

**Comment by rarkins** [comment]
Seems the best you can do is dedicated Mac minis Eg from https://www.macminivault.com/services/dedicated-mac-mini/

GitLab and Buildkite are two example solutions that have their own "runners" for such solutions. 

We may be better implementing our own "update a pod lock file" function.

**Comment by viceice** [comment]
sure, but what about transitive deps? should we rebuild the whole tree?

**Comment by rarkins** [comment]
We'd need to, if we went down that path. Maybe we can write a Ruby package that leverages pod's capabilities.

**Comment by jpolo** [comment]
Fixing this would allow to save a lot of time for react-native projects. Maintaining pod lock after an upgrade implies checkout  the branch then `yarn install`, `pod install`, `git commit` + hooks, push... 
This easily can take 10-15 minutes per MR.

**Comment by viceice** [comment]
Moving this back to ready but we will need help from comunity.

**So the current idea:**
* parse lockfile
* update dependency
* update transitve dependencies
* update checksums

This can be done in ruby (reusing pods code) or in plain typescipt / javascript. I prefer typescript, but if it's easier in ruby then go for it. This code should  replace current updateArtifacts method.

**Comment by geraintwhite** [comment]
> Fixing this would allow to save a lot of time for react-native projects. Maintaining pod lock after an upgrade implies checkout the branch then `yarn install`, `pod install`, `git commit` + hooks, push...
> This easily can take 10-15 minutes per MR.

I tried to solve this a while ago using `postUpgradeTasks` to run `pod install` but didn't get it working properly for some reason. I will try again soon and report back.

**Comment by viceice** [comment]
`postUpgradeTasks` are only available for self-hosted renovate!

**Comment by asoneji** [comment]
I see this is marked with `status: ready` what does this mean?

Any guidance or possible suggestions we should start with to fix this issue? I wonder how the Renovate GitHub app can solve this problem?

**Comment by viceice** [comment]
> Moving this back to ready but we will need help from comunity.
> 
> **So the current idea:**
> * parse lockfile
> * update dependency
> * update transitve dependencies
> * update checksums
> 
> This can be done in ruby (reusing pods code) or in plain typescipt / javascript. I prefer typescript, but if it's easier in ruby then go for it. This code should  replace current updateArtifacts method.

☝️
